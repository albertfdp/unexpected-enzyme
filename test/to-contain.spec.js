import expect from './unexpected-enzyme';
import { mount } from 'enzyme';
import React from 'react';
import PropTypes from 'prop-types';

const User = ({ id }) => <span>User {id}</span>;

User.propTypes = {
  id: PropTypes.number
};

const Fixture = () => (
  <div>
    <ul>
      <li>
        <User id={1} name="John" />
      </li>
      <li>
        <User id={2} name="Doe" />
        <User id={3} name="Mark" />
      </li>
    </ul>
  </div>
);

describe('to-contain', () => {
  it('passes when the actual matches the expected', () => {
    expect(mount(<Fixture />), 'to contain', <User id={1} />);
    expect(mount(<Fixture />), 'to contain', <User id={2} />);
  });

  it('fails when the actual does not match the expected', () => {
    expect(
      () => expect(mount(<Fixture />), 'to contain', <User id={4} />),
      'with error matching snapshot'
    );
  });

  describe('when negated', () => {
    it('passes negated when the actual does not match the expected', () => {
      expect(mount(<Fixture />), 'not to contain', <User id={4} />);
    });
  });
});
