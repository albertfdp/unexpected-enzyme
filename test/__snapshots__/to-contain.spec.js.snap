// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`to-contain <ReactElement> fails when the actual does not match the expected 1`] = `
"expected
<Fixture>
  <div>
    <ul>
      <li>
        <User id={1} name=\\"John\\">
          <span className=\\"user\\" id=\\"user-1\\">User 1</span>
        </User>
      </li>
      <li>
        <User id={2} name=\\"Doe\\">
          <span className=\\"user\\" id=\\"user-2\\">User 2</span>
        </User>
        <User id={3} name=\\"Mark\\">
          <span className=\\"user\\" id=\\"user-3\\">User 3</span>
        </User>
      </li>
    </ul>
  </div>
</Fixture>
to contain <User id={4} />

the best match was
<Fixture // should be <User
   // missing id={4}
/>"
`;

exports[`to-contain <ReactElement> when negated fails when the actual does match the expected 1`] = `
"expected
<Fixture>
  <div>
    <ul>
      <li>
        <User id={1} name=\\"John\\">
          <span className=\\"user\\" id=\\"user-1\\">User 1</span>
        </User>
      </li>
      <li>
        <User id={2} name=\\"Doe\\">
          <span className=\\"user\\" id=\\"user-2\\">User 2</span>
        </User>
        <User id={3} name=\\"Mark\\">
          <span className=\\"user\\" id=\\"user-3\\">User 3</span>
        </User>
      </li>
    </ul>
  </div>
</Fixture>
not to contain <User id={2} />

but found the following match
<Fixture // should be <User
   // missing id={2}
/>"
`;

exports[`to-contain <selector> fails when the actual does not match the expected 1`] = `
"expected
<Fixture>
  <div>
    <ul>
      <li>
        <User id={1} name=\\"John\\">
          <span className=\\"user\\" id=\\"user-1\\">User 1</span>
        </User>
      </li>
      <li>
        <User id={2} name=\\"Doe\\">
          <span className=\\"user\\" id=\\"user-2\\">User 2</span>
        </User>
        <User id={3} name=\\"Mark\\">
          <span className=\\"user\\" id=\\"user-3\\">User 3</span>
        </User>
      </li>
    </ul>
  </div>
</Fixture>
to contain '#user-4'"
`;
